### PR TITLE
[Pipelines] Upgrade to 1.8.1

### PR DESCRIPTION
**Pull request by TomerShor**
### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove irrelevant fields.]
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[mychartname]`)

### Description:
Update pipelines charts according to https://github.com/kubeflow/pipelines/tree/25dfe6065dfce449f8ebe6109109d9626dff3a60/manifests
Including updating the relevant images and pushing them to the iguazio gcr repo.

Requirement: https://jira.iguazeng.com/browse/IG-18170

### CODE REVIEW

File: stable/pipelines/Chart.yaml [hunk 1]
```diff
@@ -1,14 +1,14 @@
 apiVersion: v1
-appVersion: ">=1.0.1"
-version: 0.9.0
+appVersion: ">=1.8.1"
+version: 0.8.0
```
Comment (liranbg) [inline stable/pipelines/Chart.yaml]: undo change + set to `0.9.1`

File: stable/pipelines/Chart.yaml [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
 name: pipelines
 description: Kubeflow pipelines framework for machine learning
 home: https://www.kubeflow.org/
 icon: https://github.com/kubeflow/marketing-materials/blob/master/logos/Raster/Kubeflow-Logo-RGB.png
 sources:
   - https://github.com/kubeflow/pipelines
 maintainers:
-  - name: Oded Messer
-    email: odedm@iguazio.com
+  - name: Tomer Shor
```
Comment (liranbg) [inline stable/pipelines/Chart.yaml:11]: 👏🏼

File: stable/pipelines/templates/argo/workflow-controller-deployment.yaml [hunk 3]
```diff
@@ -42,13 +43,31 @@ spec:
             fieldRef:
               apiVersion: v1
               fieldPath: metadata.namespace
+        - name: LEADER_ELECTION_IDENTITY
+          valueFrom:
+            fieldRef:
+              apiVersion: v1
+              fieldPath: metadata.namespace
```
Comment (liranbg) [inline stable/pipelines/templates/argo/workflow-controller-deployment.yaml:50]: mind explain what's that?

File: stable/pipelines/templates/argo/workflow-controller-deployment.yaml [hunk 4]
```diff
@@ -42,13 +43,31 @@ spec:
...
         image: {{ .Values.images.workflowController.repository }}:{{ .Values.images.workflowController.tag }}
         imagePullPolicy: {{ .Values.images.imagePullPolicy }}
         name: workflow-controller
         resources:
 {{ toYaml .Values.resources.workflowController | indent 10 }}
         terminationMessagePath: /dev/termination-log
         terminationMessagePolicy: File
+        ports:
+          - name: metrics
+            containerPort: 9090
```
Comment (liranbg) [inline stable/pipelines/templates/argo/workflow-controller-deployment.yaml:60]: nice. that means we can add a grafana dashboard to make some nice graphs

File: stable/pipelines/templates/metadata/envoy-deployment.yaml [hunk 5]
```diff
@@ -16,6 +16,8 @@ spec:
       labels:
         component: metadata-envoy
 {{ include "pipelines.commonLabels" . | indent 8 }}
+      annotations:
+        sidecar.istio.io/inject: "false"  # TOMER - NEW - check if can put in values.yaml
```
Comment (liranbg) [inline stable/pipelines/templates/metadata/envoy-deployment.yaml]: what's the default? why do we need it at all?

File: stable/pipelines/templates/metadata/grpc-deployment.yaml [hunk 6]
```diff
@@ -19,7 +19,7 @@ spec:
         component: metadata-grpc-server
 {{ include "pipelines.commonLabels" . | indent 8 }}
     spec:
-      initContainers:
+      initContainers: # TOMER - this whole section doesn't exist
```
Comment (liranbg) [inline stable/pipelines/templates/metadata/grpc-deployment.yaml]: what?

File: stable/pipelines/templates/metadata/grpc-serviceaccount.yaml [hunk 7]
```diff
@@ -0,0 +1,4 @@
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: metadata-grpc-server
```
Comment (liranbg) [inline stable/pipelines/templates/metadata/grpc-serviceaccount.yaml]: newline

File: stable/pipelines/templates/ml-pipeline/apiserver/deployment.yaml [hunk 8]
```diff
@@ -37,6 +39,8 @@ spec:
           value: "true"
         - name: OBJECTSTORECONFIG_FORCEV2SIGNATURE
           value: "true"
+        - name: AUTO_UPDATE_PIPELINE_DEFAULT_VERSION
```
Comment (liranbg) [inline stable/pipelines/templates/ml-pipeline/apiserver/deployment.yaml:40]: 👀 ?

File: stable/pipelines/templates/ml-pipeline/ui/deployment.yaml [hunk 9]
```diff
@@ -17,6 +17,8 @@ spec:
       labels:
         component: ml-pipeline-ui
 {{ include "pipelines.commonLabels" . | indent 8 }}
+      annotations:
+        cluster-autoscaler.kubernetes.io/safe-to-evict: "true"
```
Comment (liranbg) [inline stable/pipelines/templates/ml-pipeline/ui/deployment.yaml]: do we need that annotation?

File: stable/pipelines/templates/ml-pipeline/viewer-crd/deployment.yaml [hunk 10]
```diff
@@ -17,10 +17,12 @@ spec:
       labels:
         component: ml-pipeline-viewer-crd
 {{ include "pipelines.commonLabels" . | indent 8 }}
+      annotations:
+        cluster-autoscaler.kubernetes.io/safe-to-evict: "true"
     spec:
       containers:
       - image: {{ .Values.images.viewerCrdController.repository }}:{{ .Values.images.viewerCrdController.tag }}
-        imagePullPolicy: {{ .Values.images.imagePullPolicy }}
+        imagePullPolicy: Always
```
Comment (liranbg) [inline stable/pipelines/templates/ml-pipeline/viewer-crd/deployment.yaml]: no

File: stable/pipelines/values.yaml [hunk 11]
```diff
@@ -46,47 +46,54 @@ configurations:
...
-    tag: v2.7.5-igz-license-compliance
+    tag: v3.2.3-igz-license-compliance
   workflowController:
     repository: gcr.io/iguazio/workflow-controller
-    tag: v2.7.5-igz-license-compliance
+    tag: v3.2.3-igz-license-compliance
   apiServer:
-    repository: gcr.io/iguazio/ml-pipeline-api-server
-    tag: 1.0.1-igz
+    repository: gcr.io/iguazio/api-server
```
Comment (liranbg) [inline stable/pipelines/values.yaml]:
```suggestion
    repository: gcr.io/iguazio/ml-pipeline-api-server
```

**Review by liranbg — CHANGES_REQUESTED** [review]
Great work! some comment as it looks WIP

File: stable/index.yaml [hunk 12]
```diff
@@ -1,3 +0,0 @@
-apiVersion: v1
-entries: ~
-generated: 2018-01-30T17:09:13.202782+02:00
```
Comment (quaark) [inline stable/index.yaml:3]: Don't remove this :)

File: stable/pipelines/templates/ml-pipeline/persistenceagent/deployment.yaml [hunk 13]
```diff
@@ -29,6 +31,8 @@ spec:
                 fieldPath: metadata.namespace
           - name: TTL_SECONDS_AFTER_WORKFLOW_FINISH
             value: "86400"
+          - name: NUM_WORKERS
+            value: "2"
```
Comment (quaark) [inline stable/pipelines/templates/ml-pipeline/persistenceagent/deployment.yaml]: Maybe also something to add to the values file?

**Review by quaark — CHANGES_REQUESTED** [review]
Just a couple extras to what @liranbg commented

File: stable/pipelines/templates/argo/workflow-controller-deployment.yaml (hunk 3 above)
Comment (TomerShor) [reply stable/pipelines/templates/argo/workflow-controller-deployment.yaml:50]: It's the ID used for workflow controllers to elect a leader.
Ref: https://github.com/argoproj/argo-workflows/issues/4761#issuecomment-752188244

File: stable/pipelines/templates/metadata/grpc-deployment.yaml (hunk 6 above)
Comment (TomerShor) [reply stable/pipelines/templates/metadata/grpc-deployment.yaml]: 🦥

File: stable/pipelines/templates/ml-pipeline/ui/deployment.yaml (hunk 9 above)
Comment (TomerShor) [reply stable/pipelines/templates/ml-pipeline/ui/deployment.yaml]: It enables scale-down, so I removed it from all deployment yamls.

**Review by liranbg — APPROVED** [review]

**Pull request merged**